src/hooks/usePortal.ts

```diff
@@ -0,0 +1,45 @@
+import React, { useCallback, useEffect, useRef, useState } from "react"
+
+/**
+ * @param portalId - unique id to represent the portal element (as child of `<body>`)
+ * @param closeFn - function to call when ESC key is handled
+ * @param liveAndAtomic - set to true for ARIA announcements
+ */
+export default function usePortal(
```

Comment:
Could you add to the description of this hook what the intention of the hook is and when to use it? I'm not 100% sure why this is needed